src/api-transition-helpers.coffee

```diff
@@ -0,0 +1,17 @@
+LineNumberGutterName = 'line-numbers'
+
+# Given decorationParams/Properties from the 1.0 API, applies changes to enable
+# the 2.0 API under the covers. In 2.0, the `trueType` property will become `type`.
+translateDecorationParamsOldToNew = (decorationParams) ->
```

Comment:
> I think it makes sense to break this out into a separate method, because this logic is needed in two places: TextBuffer::decorateMarker, and Decoration::setProperties.

Couldn't you perform the initial translation in the `Decoration` constructor rather than in `TextEditor::decorateMarker`?